### PR TITLE
Python: Fix a bunch of QL warnings

### PR DESCRIPTION
**Pull request by tausbn**
Does what it says on the tin. Most of the remaining warnings fall into a few classes:

- Alerts relating to deprecated classes (primarily stuff relating to the old data-flow library). I can fix these, but it didn't seem like much of a priority.
- Uses of db-types outside of `lib`. Some of these should probably just be deleted (SVN support, anyone?), but may need a formal deprecation first.

All of the changes in this PR should be fairly benign, however.

---

As this doesn't change the API, I don't think a change note is required.

### CODE REVIEW

**Review by michaelnebel — APPROVED** [review]
C#: LGTM

**Review by atorralba — APPROVED** [review]
Java 👍

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 1]
```diff
@@ -15,15 +15,15 @@ private module Aiomysql {
   private import semmle.python.internal.Awaited
 
   /**
-   * A `ConectionPool` is created when the result of `aiomysql.create_pool()` is awaited.
+   * Gets the `ConnectionPool` that is created when the result of `aiomysql.create_pool()` is awaited.
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:18]:
```suggestion
   * Gets a `ConnectionPool` that is created when the result of `aiomysql.create_pool()` is awaited.
```
This being an API node, I think we should express that multiple results are likely.

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 2]
```diff
@@ -15,15 +15,15 @@ private module Aiomysql {
...
    * See https://aiomysql.readthedocs.io/en/stable/pool.html
    */
   API::Node connectionPool() {
     result = API::moduleImport("aiomysql").getMember("create_pool").getReturn().getAwaited()
   }
 
   /**
-   * A `Connection` is created when
+   * Gets the `Connection` that is created when
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:26]:
```suggestion
   * Gets a `Connection` that is created when
```

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 3]
```diff
@@ -35,7 +35,7 @@ private module Aiomysql {
   }
 
   /**
-   * A `Cursor` is created when
+   * Gets the `Cursor` that is created when
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:38]:
```suggestion
   * Gets a `Cursor` that is created when
```

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 4]
```diff
@@ -98,13 +98,13 @@ private module Aiomysql {
   }
 
   /**
-   * A `SAConnection` is created when the result of calling `aquire` on an `Engine` is awaited.
+   * Gets the `SAConnection` that is created when the result of calling `aquire` on an `Engine` is awaited.
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:101]:
```suggestion
   * Gets an `SAConnection` that is created when the result of calling `aquire` on an `Engine` is awaited.
```

File: python/ql/lib/semmle/python/frameworks/Aiopg.qll [hunk 5]
```diff
@@ -15,15 +15,15 @@ private module Aiopg {
   private import semmle.python.internal.Awaited
 
   /**
-   * A `ConectionPool` is created when the result of `aiopg.create_pool()` is awaited.
+   * Gets the `ConnectionPool` that is created when the result of `aiopg.create_pool()` is awaited.
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiopg.qll:18]:
```suggestion
   * Gets a `ConnectionPool` that is created when the result of `aiopg.create_pool()` is awaited.
```

File: python/ql/lib/semmle/python/frameworks/Aiopg.qll [hunk 6]
```diff
@@ -15,15 +15,15 @@ private module Aiopg {
...
    * See https://aiopg.readthedocs.io/en/stable/core.html#pool
    */
   API::Node connectionPool() {
     result = API::moduleImport("aiopg").getMember("create_pool").getReturn().getAwaited()
   }
 
   /**
-   * A `Connection` is created when
+   * Gets the `Connection` that is created when
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiopg.qll:26]:
```suggestion
   * Gets a `Connection` that is created when
```

File: python/ql/lib/semmle/python/frameworks/Aiopg.qll [hunk 7]
```diff
@@ -35,7 +35,7 @@ private module Aiopg {
   }
 
   /**
-   * A `Cursor` is created when
+   * Gets the `Cursor` that is created when
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiopg.qll:38]:
```suggestion
   * Gets a `Cursor` that is created when
```

File: python/ql/lib/semmle/python/frameworks/Aiopg.qll [hunk 8]
```diff
@@ -94,13 +94,13 @@ private module Aiopg {
   }
 
   /**
-   * A `SAConnection` is created when the result of calling `aquire` on an `Engine` is awaited.
+   * Gets the `SAConnection` that is created when the result of calling `aquire` on an `Engine` is awaited.
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Aiopg.qll:97]:
```suggestion
   * Gets an `SAConnection` that is created when the result of calling `aquire` on an `Engine` is awaited.
```

File: python/ql/lib/semmle/python/frameworks/Asyncpg.qll [hunk 9]
```diff
@@ -12,13 +12,13 @@ private import semmle.python.ApiGraphs
 private module Asyncpg {
   private import semmle.python.internal.Awaited
 
-  /** A `ConectionPool` is created when the result of `asyncpg.create_pool()` is awaited. */
+  /** Gets the `ConnectionPool` that is created when the result of `asyncpg.create_pool()` is awaited. */
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Asyncpg.qll:15]:
```suggestion
  /** Gets a `ConnectionPool` that is created when the result of `asyncpg.create_pool()` is awaited. */
```

File: python/ql/lib/semmle/python/frameworks/Asyncpg.qll [hunk 10]
```diff
@@ -12,13 +12,13 @@ private import semmle.python.ApiGraphs
...
   API::Node connectionPool() {
     result = API::moduleImport("asyncpg").getMember("create_pool").getReturn().getAwaited()
   }
 
   /**
-   * A `Connection` is created when
+   * Gets the `Connection` that is created when
```
Comment (yoff) [inline python/ql/lib/semmle/python/frameworks/Asyncpg.qll:21]:
```suggestion
   * Gets a `Connection` that is created when
```

**Review by yoff — CHANGES_REQUESTED** [review]
Thank you for doing this!

File: python/ql/src/Imports/DeprecatedModule.ql [hunk 11]
```diff
@@ -13,6 +13,9 @@
 import python
 
 /**
+ * Gets the name of a deprecated module (for a particular version of Python) and the name of a suggested
+ * replacement.
+ *
  * The module `name` was deprecated in Python version `major`.`minor`,
  * and module `instead` should be used instead (or `instead = "no replacement"`)
  */
```
[the remark this answers is not in the document]
Comment (tausbn) [reply python/ql/src/Imports/DeprecatedModule.ql:21]: Ah, good catch.

**Review by MathiasVP — APPROVED** [review]
C++ 👍

**Review by yoff — APPROVED** [review]
Happy to merge this now.
The `Field only used in CharPred`-warnings are interesting, but can be tackled later.

**Review by alexrford — APPROVED** [review]
Ruby 👍

**Pull request merged**